LibVLCSharp/Shared/MediaSource.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace LibVLCSharp.Shared
+{
+    /// <summary>
+    /// Source for VideoView that encapsulates the <see cref="LibVLC"/> and <see cref="MediaPlayer"/> instances />
+    /// </summary>
+    public class MediaSource : ISource
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="MediaSource"/> class
+        /// </summary>
+        /// <param name="cliOptions">command line options (https://wiki.videolan.org/VLC_command-line_help/)</param>
+        protected MediaSource(params string[] cliOptions)
+        {
+            if (CoreInitialized.Value)
+            {
+                LibVLC = new LibVLC(cliOptions);
+                MediaPlayer = new MediaPlayer(LibVLC);
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="MediaSource"/> class
+        /// </summary>
+        /// <param name="uri">URI of the media to play</param>
+        /// <param name="cliOptions">command line options (https://wiki.videolan.org/VLC_command-line_help/)</param>
+        protected MediaSource(string uri, params string[] cliOptions) : this(cliOptions)
+        {
+            MediaPlayer.Media = new Media(LibVLC, uri, Media.FromType.FromLocation);
+        }
+
+        /// <summary>
+        /// Finalizer
+        /// </summary>
+        ~MediaSource()
+        {
+            Dispose();
+        }
+
+        private static Lazy<bool> CoreInitialized { get; } = new Lazy<bool>(() =>
```

Comment:
I believe this property belongs in `Core.cs` (and it could be called multiple times, which is a case not currently handled right now but not in the scope of this PR, which is already quite big). Not quite sure about your `Lazy` choice there though.

Reply:
You are totally right. I'm going to move it to `Core.cs`. I will not need such a property in `Core` class, I can use the static constructor.

Reply:
A static constructor doesn't feel sane to me. Can you use a `IsInitialized` field, paired with the `Core.Initialize()` method, that will check that the core is not already initialized before doing anything else?

Reply:
I do not see the problem with a static constructor but I can add a thread safe IsInitialized property.
In my mind, I was going to do a static constructor that do the initialization and the Core.Initialize method was going to be empty (just to call the static constructor). It's just an implementation problem but the result is the same : we have to be able to call Core.Initialize several times and do nothing if the initialization is already done.